Lib/fontTools/ttLib/ttFont.py

```diff
@@ -666,7 +666,17 @@ def getGlyphNameMany(self, lst):
     def getGlyphID(self, glyphName):
         """Returns the ID of the glyph with the given name."""
         try:
-            return self.getReverseGlyphMap()[glyphName]
+            gid = None
+            try:
+                gid = self.getReverseGlyphMap()[glyphName]
+            except KeyError:
+                if len(self.getGlyphOrder()) == len(self._reverseGlyphOrderDict):
+                    raise
+            if gid is not None and self.getGlyphName(gid) == glyphName:
+                return gid
+            self._buildReverseGlyphOrderDict()
+            return self._reverseGlyphOrderDict[glyphName]
+
         except KeyError:
             if glyphName[:5] == "glyph":
```

Comment:
wouldn't this patch now slow down the "virtual GID" code path? Glyphs named e.g. 'glyph00123' would initially raise KeyError because they are absent by definition (hence 'virtual') and thus trigger a rebuild of the glyph map cache each single time, and only after the second KeyError their name's 5-character tail would be turned into an integer.